core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -960,19 +962,21 @@ private[spark] class TaskSetManager(
       assert (null != failureReason)
       taskSetExcludelistHelperOpt.foreach(_.updateExcludedForFailedTask(
         info.host, info.executorId, index, failureReason))
-      numFailures(index) += 1
-      if (numFailures(index) >= maxTaskFailures) {
-        logError("Task %d in stage %s failed %d times; aborting job".format(
-          index, taskSet.id, maxTaskFailures))
-        abort("Task %d in stage %s failed %d times, most recent failure: %s\nDriver stacktrace:"
-          .format(index, taskSet.id, maxTaskFailures, failureReason), failureException)
-        return
+      if (!successful(index)) {
+        numFailures(index) += 1
```

Comment:
AS I mentioned above, I think we should still count task failures no matter `successful(index)` is true or not.

Reply:
I rollback to origin code, thanks